src/main/kotlin/org/rust/ide/annotator/fixes/CreateLifetimeParameterFromUsageFix.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RsLifetime
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsGenericDeclaration
+import org.rust.lang.core.psi.ext.RsNameIdentifierOwner
+import org.rust.lang.core.psi.ext.ancestorOrSelf
+
+class CreateLifetimeParameterFromUsageFix(lifetime: RsLifetime) : LocalQuickFixAndIntentionActionOnPsiElement(lifetime) {
+    override fun getFamilyName(): String {
+        return "Create lifetime parameter"
+    }
+
+    override fun getText(): String = familyName
+
+    override fun invoke(project: Project, file: PsiFile, editor: Editor?, startElement: PsiElement, endElement: PsiElement) {
+        val context = gatherContext(startElement) ?: return
+        insertLifetime(context.declaration, project, context.lifetime)
+    }
+
+    private fun insertLifetime(declaration: RsGenericDeclaration, project: Project, startElement: RsLifetime) {
```

Comment:
Looks like `startElement` is a temporal name and should be renamed to `lifetime` or something